st2api/st2api/controllers/v1/actionexecutions.py

```diff
@@ -277,17 +294,35 @@ def validate(self):
 
             return self
 
-    @jsexpose(body_cls=ExecutionSpecificationAPI, status_code=http_client.CREATED)
-    def post(self, spec_api, execution_id, no_merge=False):
+    def post(self, spec_api, id, requester_user=None, no_merge=False, **kwargs):
         """
         Re-run the provided action execution optionally specifying override parameters.
 
         Handles requests:
 
             POST /executions/<id>/re_run
         """
+
+        if (spec_api.tasks or spec_api.reset) and spec_api.parameters:
```

Comment:
At that point I wasn't sure I'll be using models anymore. I'm still not sure about that, honestly, I don't like us doing stuff like https://github.com/StackStorm/st2/blob/master/st2common/st2common/models/api/action.py#L109 and I see an explicit controller with no hidden logic as the way to achieve that. I do not plan to fix every problem in this PR, though, it's huge already.